eras/alonzo/impl/src/Cardano/Ledger/Alonzo/Scripts.hs

```diff
@@ -428,8 +442,8 @@ instance Era era => FromCBOR (Annotator (Script era)) where
 validScript :: ProtVer -> Script era -> Bool
 validScript pv scrip = case scrip of
   TimelockScript sc -> deepseq sc True
-  PlutusScript PlutusV1 bytes -> PV1.isScriptWellFormed (transProtocolVersion pv) bytes
-  PlutusScript PlutusV2 bytes -> PV2.isScriptWellFormed (transProtocolVersion pv) bytes
+  PlutusScript PlutusV1 bytes -> (not . null) (PV1.assertScriptWellFormed (transProtocolVersion pv) bytes :: Either ScriptDecodeError ())
+  PlutusScript PlutusV2 bytes -> (not . null) (PV2.assertScriptWellFormed (transProtocolVersion pv) bytes :: Either ScriptDecodeError ())
```

Comment:
It is better to use specialized function `isRight` instead of relying on Foldable instance for Either: